apollo-federation/src/operation/tests/mod.rs

```diff
@@ -21,30 +21,59 @@ use crate::schema::position::ObjectTypeDefinitionPosition;
 macro_rules! assert_normalized {
     ($schema_doc: expr, $query: expr, @$expected: literal) => {{
         let schema = parse_schema($schema_doc);
-        let without_fragments = parse_and_expand(&schema, $query).unwrap();
+        let without_fragments = parse_and_expand(&schema, $query).expect("operation is valid and can be normalized");
         insta::assert_snapshot!(without_fragments, @$expected);
         without_fragments
     }};
 }
 
+macro_rules! assert_normalized_equal {
+    ($schema_doc: expr, $query: expr, $expected: literal) => {{
+        let normalized = assert_normalized!($schema_doc, $query, @$expected);
+
+        let schema = parse_schema($schema_doc);
+        let original_document = ExecutableDocument::parse_and_validate(schema.schema(), $query, "query.graphql").expect("valid document");
+        let normalized_document = normalized.clone().try_into().expect("valid normalized document");
+        compare_operations(&schema, &original_document, &normalized_document).expect("documents A and B are the same");
+        compare_operations(&schema, &normalized_document, &original_document).expect("documents B and A are the same");
+        normalized
+    }};
+}
+
+macro_rules! assert_equal_ops {
+    ($schema: expr, $first: expr, $second: expr) => {
+        let original_document: Valid<ExecutableDocument> =
+            $first.try_into().expect("valid document");
+        let minified_document: Valid<ExecutableDocument> =
+            $second.try_into().expect("valid document");
+        compare_operations($schema, &original_document, &minified_document)
+            .expect("documents are the same");
```

Comment:
Similarly to the above comment, `expect("The first document includes the second")`.